atomics/include/desul/atomics/Atomic_Ref.hpp

```diff
@@ -79,6 +79,10 @@ class AtomicRef {
   DESUL_IMPL_DEFINE_ATOMIC_FETCH_OP(xor)
   DESUL_IMPL_DEFINE_ATOMIC_COMPOUND_ASSIGNMENT_OP(^=, xor)
   DESUL_IMPL_DEFINE_ATOMIC_FETCH_OP(nand)
```

Comment:
I think we should have them take the operand as unsigned int instead of T

Reply:
isn't that true for mod too?

Reply:
I am talking about consistency with the free function signatures
https://github.com/desul/desul/blob/60c1115af231f4bfe0d70a3e0cf54b2cc0ff1594/atomics/include/desul/atomics/Fetch_Op_Generic.hpp#L57-L60
We currently define `mod` as taking the operand as a `T`.